references/references.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2023 Nephio.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package references
+
+
+// NamespaceReference is used to reference a particular krm intstance
+type NamespaceReference struct {
+	// Namespace is the namespace of the referent.
+	// This field is required when referring to a Namespace-scoped resource and
+	// MUST be unset when referring to a Cluster-scoped resource.
+	//
+	// +optional
+	Namespace *Namespace `json:"namespace,omitempty"`
+
+	// Name is unique within a namespace to reference a nfClass.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name"`
+}
+
+// PackageRevisionReference is used to reference a particular package revision.
+type PackageRevisionReference struct {
```

Comment:
Not sure this should be here. We also have this: https://github.com/GoogleContainerTools/kpt/blob/porch/v0.0.15/porch/controllers/packagevariantsets/api/v1alpha1/packagevariantset_types.go#L56

Reply:
ok will remove